src/elm/Search.elm

```diff
@@ -0,0 +1,478 @@
+module Search exposing
+    ( ActiveTab(..)
+    , Model
+    , Msg
+    , State(..)
+    , closeSearch
+    , init
+    , isActive
+    , subscriptions
+    , update
+    , viewEmptyResults
+    , viewForm
+    , viewOffers
+    , viewRecentQueries
+    , viewResultsOverview
+    , viewTabs
+    )
+
+import Action exposing (Action)
+import Api.Graphql
+import Cambiatus.Object
+import Cambiatus.Object.Product
+import Cambiatus.Object.SearchResult
+import Cambiatus.Query
+import Eos exposing (Symbol)
+import Graphql.Http
+import Graphql.OptionalArgument exposing (OptionalArgument(..))
+import Graphql.SelectionSet as SelectionSet exposing (SelectionSet, with)
+import Html exposing (Html, br, button, div, h3, img, input, li, p, span, strong, text, ul)
+import Html.Attributes exposing (class, disabled, minlength, placeholder, required, src, type_, value)
+import Html.Events exposing (onClick, onFocus, onInput, onSubmit)
+import Icons
+import Json.Decode as Decode exposing (list, string)
+import Json.Encode as Encode
+import List.Extra as List
+import Ports
+import Route exposing (Route)
+import Session.Shared exposing (Shared, Translators)
+
+
+
+-- MODEL
+
+
+type alias Model =
+    { state : State
+    , currentQuery : String
+    , recentQueries : List String
+    , selectedCommunity : Symbol
+    }
+
+
+init : Symbol -> Model
+init selectedCommunity =
+    { state = Inactive
+    , currentQuery = ""
+    , recentQueries = []
+    , selectedCommunity = selectedCommunity
+    }
+
+
+
+-- TYPES
+
+
+type State
+    = Inactive
+    | Loading
+    | RecentSearchesShowed
+    | ResultsShowed (Maybe SearchResult) ActiveTab
```

Comment:
do you think here is a good place to add remote data?

Reply:
I think the `RemoteData` pattern fits here for the `ResultsShowed` variant since it's responsible for the actual data fetching. I tried it and I like the changes it brought! I made some refactoring to make the `RemoteData` rule the fetching search results process. Thanks for the idea! Hope you like the changes too :)